tsl/src/remote/dist_ddl.c

```diff
@@ -343,6 +343,19 @@ dist_ddl_state_get_hypertable_type(void)
 	return type;
 }
 
+static void
+dist_ddl_process_create_schema(ProcessUtilityArgs *args)
+{
+	CreateSchemaStmt *stmt = castNode(CreateSchemaStmt, args->parsetree);
+
+	if (dist_util_membership() != DIST_MEMBER_ACCESS_NODE)
```

Comment:
Why do we need this check here? 

I don't think we should even get to this point if we are not an access node since we shouldn't process dist ddl on data nodes. This should have been handled much higher up in the call-stack IMO.

Also, I don't see this check in other dist_ddl_process_ functions, so it seems inconsistent.

I think maybe Asserting is OK.

Reply:
This can be triggered for a regular database, not only a data node

Reply:
Yeah, we shouldn't process dist_util stuff at all for anything but an access node. So, having the check at this low level is confusing.